### PR TITLE
CMake build scripts for native linux-ppc32

### PR DESCRIPTION
**Pull request by shingarov**
PowerPC 32-bit compiler mostly works for me when cross-built using
the autotools-based build system.  However, this latter build sys-
tem does not work for JitBuilder, nor for TrilTests -- which are
essential for work on the compiler.

This commit enables CMake on PPC32 Linux (native, not cross).

Signed-off-by: Boris Shingarov <shingarov@labware.com>

### CODE REVIEW

**Comment by 0xdaryl** [comment]
@ymanton , would you mind having a look at these options to make sure they make sense?

**Comment by fjeremic** [comment]
@dnakamura FYI as well in case you wish to review.

File: cmake/modules/platform/toolcfg/gnu.cmake [hunk 1]
```diff
@@ -40,16 +40,18 @@ if(OMR_ENV_DATA64)
...
-		-m32
-		-msse2
-	)
-	list(APPEND OMR_PLATFORM_EXE_LINKER_OPTIONS
-		-m32
-	)
-	list(APPEND OMR_PLATFORM_SHARED_LINKER_OPTIONS
-		-m32
-	)
+	if (NOT (CMAKE_SYSTEM_PROCESSOR MATCHES "ppc"))
```
Comment (dnakamura) [inline cmake/modules/platform/toolcfg/gnu.cmake]: we should be testing against our "normalized" variables. IE `if(NOT OMR_ARCH_POWER)`

File: cmake/modules/platform/toolcfg/gnu.cmake [hunk 2]
```diff
@@ -40,16 +40,18 @@ if(OMR_ENV_DATA64)
...
+			-m32
+			-msse2
+		)
+		list(APPEND OMR_PLATFORM_EXE_LINKER_OPTIONS
+			-m32
+		)
+		list(APPEND OMR_PLATFORM_SHARED_LINKER_OPTIONS
+			-m32
+		)
+	endif()
```
[the remark this answers is not in the document]
Comment (shingarov) [reply cmake/modules/platform/toolcfg/gnu.cmake]: Thanks.

File: cmake/modules/platform/toolcfg/gnu.cmake [hunk 3]
```diff
@@ -38,18 +38,18 @@ if(OMR_ENV_DATA64)
 		list(APPEND OMR_PLATFORM_SHARED_LINKER_OPTIONS
 			-m64
 		)
+	elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "x86|i[3-6]86")
```
Comment (ymanton) [inline cmake/modules/platform/toolcfg/gnu.cmake]: Is this check needed now that you're checking `if(OMR_ARCH_X86)` for the entire block?

If so, `OMR_ENV_DATA32` might be a better variable to check against, that way it reads as

```
if x86:
  if 64-bit build: "-m64"
  else if 32-bit build: "-m32 -msse2"
```

and should also work if you're on 64-bit machine but want to do a 32-bit build.

Comment (shingarov) [reply cmake/modules/platform/toolcfg/gnu.cmake]: I think you are right, there are no other scenarios on the OMR_ARCH_X86 architecture other than i386 and and64.

Comment (ymanton) [reply cmake/modules/platform/toolcfg/gnu.cmake]: No magical secret that I know of, I don't know of anyone who's tried cross or is actively working on it at the moment.

Comment (shingarov) [reply cmake/modules/platform/toolcfg/gnu.cmake]: @ymanton 
> LGTM
Does anything else need to happen on my side before this PR gets merged?

Comment (ymanton) [reply cmake/modules/platform/toolcfg/gnu.cmake]: I'm OK with the changes. @0xdaryl?

**Comment by 0xdaryl** [comment]
@genie-omr build all

**Comment by shingarov** [comment]
@genie-omr build all

**Comment by 0xdaryl** [comment]
@genie-omr build all

**Pull request merged**